src/braket/default_simulator/openqasm/circuit.py

```diff
@@ -63,28 +64,47 @@ def num_qubits(self) -> int:
 
     @property
     def basis_rotation_instructions(self):
-        """
-        This function assumes all observables are commuting.
-        """
-
         basis_rotation_instructions = []
-        measured_qubits = set()
+        observable_map = {}
+
+        def process_observable(observable):
```

Comment:
Why a nested function instead of a private function?

Reply:
This way it has access to observable map as a closure. I can move it out and add an arg if you're opinionated here though